source/opt/inline_pass.cpp

```diff
@@ -0,0 +1,430 @@
+    newVars->push_back(std::move(var_inst));
+  }
+
+  // Clone and map callee code
+  bool prevInstWasReturn = false;
+  uint32_t returnLabelId = 0;
+  bool multiBlocks = false;
+  std::unique_ptr<ir::BasicBlock> bp;
+  calleeFn->ForEachInst(
+      [&newBlocks, &callee2caller, &call_bi, &call_ii, &bp, &prevInstWasReturn,
+       &returnLabelId, &returnVarId, &calleeTypeId, &multiBlocks, &postCallSI,
+       &preCallSI, this](const ir::Instruction* cpi) {
+        switch (cpi->opcode()) {
+          case SpvOpFunction:
+          case SpvOpFunctionParameter:
+          case SpvOpVariable:
+            // already processed
+            break;
+          case SpvOpLabel: {
+            // if previous instruction was early return, insert branch
+            // instruction
+            // to return block
+            if (prevInstWasReturn) {
+              if (returnLabelId == 0) returnLabelId = this->TakeNextId();
+              AddBranch(returnLabelId, &bp);
+              prevInstWasReturn = false;
+            }
+            // finish current block (if it exists) and get label for next block
+            uint32_t labelId;
+            bool firstBlock = false;
+            if (bp != nullptr) {
+              newBlocks->push_back(std::move(bp));
+              // if result id is already mapped, use it, otherwise get a new
+              // one.
+              const uint32_t rid = cpi->result_id();
+              const auto mapItr = callee2caller.find(rid);
+              labelId = (mapItr != callee2caller.end()) ? mapItr->second
+                                                        : this->TakeNextId();
+            } else {
+              // first block needs to use label of original block
+              // but map callee label in case of phi reference
+              labelId = call_bi->GetLabelId();
+              callee2caller[cpi->result_id()] = labelId;
+              firstBlock = true;
+            }
+            // create first/next block
+            const std::vector<ir::Operand> label_in_operands;
+            std::unique_ptr<ir::Instruction> newLabel(
+                new ir::Instruction(SpvOpLabel, 0, labelId, label_in_operands));
+            bp.reset(new ir::BasicBlock(std::move(newLabel)));
+            if (firstBlock) {
+              // Copy contents of original caller block up to call instruction
+              for (auto cii = call_bi->begin(); cii != call_ii; cii++) {
+                std::unique_ptr<ir::Instruction> spv_inst(
+                    new ir::Instruction(*cii));
+                // remember OpSampledImages for possible regeneration
+                if (spv_inst->opcode() == SpvOpSampledImage) {
+                  auto* samp_inst_ptr = spv_inst.get();
+                  preCallSI[spv_inst->result_id()] = samp_inst_ptr;
+                }
+                bp->AddInstruction(std::move(spv_inst));
+              }
+            } else
+              multiBlocks = true;
```

Comment:
`else { ... }`. `if` and `else` should match w.r.t. `{}`. If one of them has `{}`, the other should do.